### PR TITLE
Paralize aggregate operator if there is a group by clause

### PR DESCRIPTION
**Pull request by liuyehcf**
Considering the two-level aggregation

1. `select count(*) from t_0`. In this case, the second level aggregation cannot paralize
2. `select count(*) from t_0 group by col_0`. In this case, the second level aggregation can be paralized, one of which only need process part of data, and we only need to ensure that those data segments do not intersect with each other(shuffle based on group by column)

Test info
* tpch
* scale factor: 1
* sql: `select sum(L_QUANTITY) as sum_quantity from lineitem group by L_ORDERKEY`
* result:
    * before optimize: 4.07s (instance=1, dop=0)
    * after optimize: 0.28s (instance=1, dop=0)

### CODE REVIEW

**Comment by liuyehcf** [comment]
run starrocks_be_unittest

**Review by satanson — DISMISSED** [review]

File: be/src/exec/pipeline/exchange/local_exchange.h [hunk 1]
```diff
@@ -64,7 +64,7 @@ class PartitionExchanger final : public LocalExchanger {
         LocalExchangeSourceOperatorFactory* _source;
         const bool _is_shuffle;
         // Compute per-row partition values.
-        const std::vector<ExprContext*>& _partition_expr_ctxs;
+        const std::vector<ExprContext*> _partition_expr_ctxs;
```
Comment (ZiheLiu) [inline be/src/exec/pipeline/exchange/local_exchange.h:67]: Why `&` is removed, which will cause every `PartitionExchanger` copies `std::vector<ExprContext*>` from `LocalExchanger`?

Comment (liuyehcf) [reply be/src/exec/pipeline/exchange/local_exchange.h:67]: aggregator's decompose use a local variable, so copy needed here. And the cost of copy pointers is cheap

Comment (ZiheLiu) [reply be/src/exec/pipeline/exchange/local_exchange.h:67]: Maybe we can add a member const std::vector<ExprContext*> _partition_expr_ctxs to LocalExchanger, and use reference to it in every PartitionExchanger

**Comment by ZiheLiu** [comment]
> aggregator's decompose use a local vector, so copy needed here. And the cost of copy pointers is cheap

@liuyehcf Maybe we can add a member `const std::vector<ExprContext*> _partition_expr_ctxs` to `LocalExchanger`, and use reference to it in every `PartitionExchanger`.

**Review by kangkaisen — DISMISSED** [review]

**Review by kangkaisen — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Pull request merged**